contributing-docs/testing/unit_tests.rst

```diff
@@ -1108,6 +1108,133 @@ This prepares airflow .whl package in the dist folder.
+
+This means that you can modify and run tests and re-run them, but if you want to modify provider code
+you need to exit breeze, rebuild the provider package and restart breeze using the command above.
+
+Rebuilding single provider package can be done using this command:
+
+.. code-block::bash
+
+  breeze release-management prepare-provider-packages \
+    --version-suffix-for-pypi dev0 --package-format wheel <provider>
+
+
+Note that some of the tests if written without taking care about the compatibility, might not work with older
+versions of Airflow - this is because of refactorings, renames, and tests relying on internals of Airflow that
+are not part of the public API. We deal with it in one of the following ways:
+
+1) if the whole provider is supposed to only work for later airflow version, we remove the whole provider
+   by excluding it from compatibility test configuration (see below)
+
+2) Some compatibility shims are defined in ``tests/test_utils/compat.py`` - and they can be used to make the
+   tests compatible - for example importing ``ParseImportError`` after the exception has been renamed from
+   ``ImportError`` and it would fail in Airflow 2.9, but we have a fallback import in ``compat.py`` that
+   falls back to old import automatically, so all tests testing / expecting ``ParseImportError`` should import
+   it from the ``tests.tests_utils.compat`` module. There are few other compatibility shims defined there and
+   you can add more if needed in a similar way.
+
+3) If only some tests are not compatible and use features that are available only in newer airflow version,
+    we can mark those tests with appropriate ``AIRFLOW_V_2_X_PLUS`` boolean constant defined in ``compat.py``
+    For example:
+
+.. code-block::python
+
+  from tests.test_utils.compat import AIRFLOW_V_2_7_PLUS
+
+  @pytest.mark.skip(not AIRFLOW_V_2_7_PLUS, reason="The tests should be skipped for Airflow < 2.7")
+  def some_test_that_only_works_for_airflow_2_7_plus():
+    pass
+
+4) Sometimes, the tests should only be run when airflow is installed from the sources. In this case you can
+   add conditional ``skipif`` markerfor ``RUNNING_TESTS_AGAINST_AIRFLOW_PACKAGES`` to the test. For example:
+
+.. code-block::python
+
+  @pytest.mark.skipif(RUNNING_TESTS_AGAINST_AIRFLOW_PACKAGES,
+                      reason="Plugin initialization is done early in case of packages")
+  def test_plugin():
+     pass
+
+
+How providers compatibility tests are run in CI ?
+-------------------------------------------------
+
+We run a set of back-compatibility tests based on the configuration specified in the
+``BASE_PROVIDERS_COMPATIBILITY_CHECKS`` constant in the ``./dev/breeze/src/airflow_breeze/global_constants.py``
+file - where we specify:
+* python version
+* airflow version
+* which providers should be removed (exclusions)
+* whether to run tests
+
+We deal with it in the following way:
+
+* If the whole provider depends
```

Comment:
I guess it is missing something here

Reply:
Left-over: it's explained a paragraph above that one . Good spot :)